File: In Kotlin/romantoint.kt
fun romanToInt(s: String): Int {
    val roman = mapOf(
        'I' to 1, 'V' to 5, 'X' to 10,
        'L' to 50, 'C' to 100, 'D' to 500, 'M' to 1000
    )

    var total = 0
    var prev = 0

    for (ch in s.reversed()) {
        val value = roman[ch]!!
        if (value < prev)
            total -= value
        else {
            total += value
            prev = value
        }
    }

    return total
}

fun main() {
    val s = "MCMXCIV"
    println("Roman numeral: $s → Integer: ${romanToInt(s)}")
}

File: In Ruby/romantoint.rb
def roman_to_int(s)
  roman = {
    'I' => 1, 'V' => 5, 'X' => 10,
    'L' => 50, 'C' => 100, 'D' => 500, 'M' => 1000
  }

  total = 0
  prev = 0

  s.reverse.each_char do |ch|
    value = roman[ch]
    if value < prev
      total -= value
    else
      total += value
      prev = value
    end
  end

  total
end

s = "MCMXCIV"
puts "Roman numeral: #{s} → Integer: #{roman_to_int(s)}"
